src/components/date/__internal__/date-picker/date-picker.component.tsx

```diff
@@ -60,121 +66,209 @@ const popoverMiddleware = [
   }),
 ];
 
-export const DatePicker = React.forwardRef<HTMLDivElement, DatePickerProps>(
-  (
-    {
-      inputElement,
-      minDate,
-      maxDate,
-      selectedDays,
-      disablePortal,
-      onDayClick,
-      pickerMouseDown,
-      pickerProps,
-      open,
-    }: DatePickerProps,
-    ref
-  ) => {
-    const l = useLocale();
-    const { localize, options } = l.date.dateFnsLocale();
-    const { weekStartsOn } = options || /* istanbul ignore next */ {};
-    const monthsLong = useMemo(
-      () =>
-        Array.from({ length: 12 }).map((_, i) => {
-          const month = localize?.month(i);
-          return month[0].toUpperCase() + month.slice(1);
-        }),
-      [localize]
-    );
-    const monthsShort = useMemo(
-      () =>
-        Array.from({ length: 12 }).map((_, i) =>
-          localize?.month(i, { width: "abbreviated" }).substring(0, 3)
-        ),
-      [localize]
+export const DatePicker = ({
+  inputElement,
+  minDate,
+  maxDate,
+  selectedDays,
+  disablePortal,
+  onDayClick,
+  pickerMouseDown,
+  pickerProps,
+  open,
+  setOpen,
+  pickerTabGuardId,
+}: DatePickerProps) => {
+  const l = useLocale();
+  const { localize, options } = l.date.dateFnsLocale();
+  const { weekStartsOn } = options || /* istanbul ignore next */ {};
+  const monthsLong = useMemo(
+    () =>
+      Array.from({ length: 12 }).map((_, i) => {
+        const month = localize?.month(i);
+        return month[0].toUpperCase() + month.slice(1);
+      }),
+    [localize]
+  );
+  const monthsShort = useMemo(
+    () =>
+      Array.from({ length: 12 }).map((_, i) =>
+        localize?.month(i, { width: "abbreviated" }).substring(0, 3)
+      ),
+    [localize]
+  );
+  const weekdaysLong = useMemo(
+    () => Array.from({ length: 7 }).map((_, i) => localize?.day(i)),
+    [localize]
+  );
+  const weekdaysShort = useMemo(() => {
+    const isGivenLocale = (str: string) => l.locale().includes(str);
+    return Array.from({ length: 7 }).map((_, i) =>
+      localize
+        ?.day(
+          i,
+          ["de", "pl"].some(isGivenLocale)
+            ? { width: "wide" }
+            : { width: "abbreviated" }
+        )
+        .substring(0, isGivenLocale("de") ? 2 : 3)
     );
-    const weekdaysLong = useMemo(
-      () => Array.from({ length: 7 }).map((_, i) => localize?.day(i)),
-      [localize]
-    );
-    const weekdaysShort = useMemo(() => {
-      const isGivenLocale = (str: string) => l.locale().includes(str);
-      return Array.from({ length: 7 }).map((_, i) =>
-        localize
-          ?.day(
-            i,
-            ["de", "pl"].some(isGivenLocale)
-              ? { width: "wide" }
-              : { width: "abbreviated" }
-          )
-          .substring(0, isGivenLocale("de") ? 2 : 3)
+  }, [l, localize]);
+  const ref = useRef<HTMLDivElement>(null);
```

Comment:
question: if ref is being moved internal does that make this a breaking change?

Albeit it was undocumented afaik...

Reply:
No the ref was only previously set in the date component (I don't even think it was passed to the picker), so there's no change in the exposed interface here

Reply:
I guess I missed that this was an internal component only as well. No problem!